tests/unit/core/client/test_ConnectionManager.py

```diff
@@ -0,0 +1,235 @@
+import sys
+import time
+import logging
+
+from lbrynet.core import log_support
+from lbrynet.core.client.ConnectionManager import ConnectionManager
+from lbrynet.core.client.ClientRequest import ClientRequest
+from lbrynet.core.server.ServerProtocol import ServerProtocol
+from lbrynet.core.RateLimiter import RateLimiter
+from lbrynet.core.Peer import Peer
+from lbrynet.core.PeerManager import PeerManager
+from lbrynet.core.Error import ConnectionClosedBeforeResponseError, NoResponseError
+
+from twisted.trial import unittest
+from twisted.internet import defer, reactor, task
+from twisted.internet.task import deferLater
+from twisted.internet.protocol import Protocol, ServerFactory
+from lbrynet import conf
+from lbrynet.interfaces  import IQueryHandlerFactory, IQueryHandler, IRequestCreator
+
+from zope.interface import implements
+
+# Set up logging if debugging is necessary for writing more tests
+LOG_TO_CONSOLE = False
+if LOG_TO_CONSOLE:
+    handler = logging.StreamHandler(sys.stdout)
+    logging.basicConfig()
+    logger = logging.getLogger()
+    logger.setLevel(0)
+    log_support.configure_console()
+    log_support.configure_twisted()
```

Comment:
out of scope for this PR, but we need a better way to configure logging for the tests as this snippet of code can't be on each test.  Maybe worth looking into whether trial has a way to configure logging.